mod-source-record-storage-server/src/main/java/org/folio/services/handlers/AbstractPostProcessingEventHandler.java

```diff
@@ -289,13 +302,18 @@ private void executeHridManipulation(Record record, JsonObject externalEntity) {
    * @param tenantId - tenantId
    * @return - Future with Record result
    */
-  private Future<Record> saveRecord(Record record, String tenantId) {
+  private Future<Record> saveRecord(Record record, String tenantId, String centralTenantId) {
```

Comment:
Something seems off with this method. It should only accept a Record and a tenant Id.

There shouldn't be a corresponding source record for the shadow instance in the member tenant so the source record should never be present and no need to update the parsed record. I think you should invoke the copy of the snapshot outside of this method and call it right before calling this method.

Reply:
@okolawole-ebsco  Thank you for your comment!
I agree with you.
Actually, I changed this logic and moved it to the higher level.
Currently, this logic regarding snapshots is inside my method "saveRecordForCentralTenant".
And I left saveRecord() one with the default implementation. 
Feel free to review it!

Reply:
Looks good!